src/content/1.7/modules/sample-modules/extending-sf-form-with-upload-image-field.md

```diff
@@ -0,0 +1,586 @@
+class Installer
+{
+    /**
+     * Module's installation entry point.
+     *
+     * @param Module $module
+     *
+     * @return bool
+     */
+    public function install(Module $module): bool
+    {
+        if (!$this->registerHooks($module)) {
+            return false;
+        }
+
+        if (!$this->installDatabase()) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Module's uninstallation entry point.
+     *
+     * @return bool
+     */
+    public function uninstall(): bool
+    {
+        return $this->uninstallDatabase();
+    }
+
+    /**
+     * Install the database modifications required for this module.
+     *
+     * @return bool
+     */
+    private function installDatabase(): bool
+    {
+        $queries = [
+            'CREATE TABLE IF NOT EXISTS `'._DB_PREFIX_.'supplier_extra_image` (
+              `id_extra_image` int(11) NOT NULL AUTO_INCREMENT,
+              `id_supplier` int(11) NOT NULL,
+              `image_name` varchar(64) NOT NULL,
+              PRIMARY KEY (`id_extra_image`)
+            ) ENGINE='._MYSQL_ENGINE_.' DEFAULT CHARSET=utf8;',
+        ];
+
+        return $this->executeQueries($queries);
+    }
+
+    /**
+     * Uninstall database modifications.
+     *
+     * @return bool
+     */
+    private function uninstallDatabase(): bool
+    {
+        $queries = [
+            'DROP TABLE IF EXISTS `'._DB_PREFIX_.'supplier_extra_image`',
+        ];
+
+        return $this->executeQueries($queries);
+    }
+
+    /**
+     * Register hooks for the module.
+     *
+     * @param Module $module
+     *
+     * @return bool
+     */
+    private function registerHooks(Module $module): bool
+    {
+        $hooks = [
+            'actionSupplierFormBuilderModifier',
+            'actionAfterCreateSupplierFormHandler',
+            'actionAfterUpdateSupplierFormHandler',
+        ];
+
+        return (bool) $module->registerHook($hooks);
+    }
+
+    /**
+     * A helper that executes multiple database queries.
+     *
+     * @param array $queries
+     *
+     * @return bool
+     */
+    private function executeQueries(array $queries): bool
+    {
+        foreach ($queries as $query) {
+            if (!Db::getInstance()->execute($query)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+}
+```
+
+Let's use `Installer` class inside the main module class by adding code snippet 
+below to `DemoExtendSymfonyForm2` class.
+
+```php
+    /**
+     * @return bool
+     */
+    public function install()
+    {
+        if (!parent::install()) {
+            return false;
+        }
+
+        $installer = new Installer();
+
+        return $installer->install($this);
+    }
+
+    /**
+     * @return bool
+     */
+    public function uninstall()
+    {
+        $installer = new Installer();
+
+        return $installer->uninstall() && parent::uninstall();
+    }
+```
+
+Let's create `SupplierExtraImage` entity class:
```

Comment:
```suggestion
Let's create `SupplierExtraImage` entity class. We use [Doctrine](https://devdocs.prestashop.com/1.7/modules/concepts/doctrine/) which is available for PrestaShop modules since version 1.7.6.
```